File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public ArrayList<Integer> solution(int[] progresses, int[] speeds) {
        // 1.        
        Queue<Integer> queue = new LinkedList<>();
        for(int i=0; i<progresses.length; i++){
            int day = (int)Math.ceil((100.0 - progresses[i])/speeds[i]);
            queue.add(day);
        }
        
        // 2.
        ArrayList<Integer> answer = new ArrayList<>();
        while(!queue.isEmpty()){
            int temp = 1;
            int top = queue.poll();
            while(!queue.isEmpty() && top >= queue.peek()){
                top = Math.max(top, queue.poll()); // ** top 갱신 조심
                temp++;
            }
            answer.add(temp);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math

def solution(progresses, speeds):
    days = deque([])
    for i in range(len(progresses)):
        day = math.ceil((100 - progresses[i]) / speeds[i])
        days.append(day)
    print(days)
    
    answer = []
    while days:
        front = days.popleft()
        count = 1
        while days and front >= days[0]:
            days.popleft()
            count += 1
        answer.append(count)
    return answer